File: quark/test/emit/json_building/py/json_building.py
from quark_runtime import *


def test_building():
    _println(((((_JSONObject()).setObjectItem(u"sub-object", (((((_JSONObject()).setObjectItem(u"astring", (_JSONObject()).setString(u"string value"))).setObjectItem(u"aint", (_JSONObject()).setNumber(9))).setObjectItem(u"atrue", (_JSONObject()).setBool(1))).setObjectItem(u"afalse", (_JSONObject()).setBool(0))).setObjectItem(u"anull", (_JSONObject()).setNull()))).setObjectItem(u"sub-list", (((_JSONObject()).setListItem(0, (_JSONObject()).setString(u"after this string expect 42 and null"))).setListItem(1, (_JSONObject()).setNumber(42))).setListItem(2, (_JSONObject()).setNull()))).setObjectItem(u"sub-string", (_JSONObject()).setString(u"a string"))).toString());


def main():
    test_building();


if __name__ == "__main__":
    main()

File: quarkc/test/emit/expected/rb/json_building/lib/json_building.rb
module Quark
require "quark"
def self.json_building; JsonBuilding; end
module JsonBuilding
require_relative 'json_building' # 0 () ()

def self.test_building()
    
    ::DatawireQuarkCore.print(::DatawireQuarkCore::JSONObject.new.setObjectItem("sub-object", ::DatawireQuarkCore::JSONObject.new.setObjectItem("astring", ::DatawireQuarkCore::JSONObject.new.setString("string value")).setObjectItem("aint", ::DatawireQuarkCore::JSONObject.new.setNumber(9)).setObjectItem("atrue", ::DatawireQuarkCore::JSONObject.new.setBool(true)).setObjectItem("afalse", ::DatawireQuarkCore::JSONObject.new.setBool(false)).setObjectItem("anull", ::DatawireQuarkCore::JSONObject.new.setNull())).setObjectItem("sub-list", ::DatawireQuarkCore::JSONObject.new.setListItem(0, ::DatawireQuarkCore::JSONObject.new.setString("after this string expect 42 and null")).setListItem(1, ::DatawireQuarkCore::JSONObject.new.setNumber(42)).setListItem(2, ::DatawireQuarkCore::JSONObject.new.setNull())).setObjectItem("sub-string", ::DatawireQuarkCore::JSONObject.new.setString("a string")).toString())


    nil
end

def self.call_main()
    self.main(::DatawireQuarkCore::List.new(ARGV))

    nil
end
def self.main(args)
    
    ::Quark.json_building.test_building()


    nil
end

if __FILE__ == $0 then ::Quark.json_building.call_main() end

end # module JsonBuilding
end # module Quark
